File: src/Server Data/Data/Scripts/aggressive_npc.lua
import('Lunar.Core')
import('Lunar.Core.Utilities')
import('Lunar.Core.Utilities.Data')
import('Lunar.Core.Utilities.Logic')
import("System")
import("Lunar.Server")
import("Lunar.Server.Utilities")
import("Lunar.Server.World.Actors")
import("Lunar.Server.World.BehaviorDefinition")
import("Lunar.Server.Utilities.Scripting")

require("Data.Scripts.npc_common")

function Update(args)
	acquireTarget(args.Invoker, args[0])
end

function OnCreated(args)
	randomWalkTmr = GameTimer(1000)

	GameTimerManager.Instance:Register("randomWalkTmr", randomWalkTmr)
end

function Attack(args)
	return 10
end

File: src/Lunar.Server/Server Data/Scripts/aggressive_npc.py
import sys
import clr
clr.AddReference('Lunar.Core')
clr.AddReference('Lunar.Server')
clr.AddReference('System')
import npc_common
from Lunar.Server.Utilities import *
from Lunar.Server.World.BehaviorDefinition import *


class AggressiveNPCBehaviorDefinition(ActorBehaviorDefinition):
	def __init__(self):
		return 0
	
	def Update(self, npc, gameTime):
		npc_common.acquire_target(npc, gameTime)

	def OnCreated(self, npc):
		print("NPC " + str(npc.Descriptor.Name) + " aggressive behavior handler created...")
		randomWalkTmr = GameTimer(500)
		npc.GameTimerManager.Register("randomWalkTmr" + str(npc.GetHashCode()), randomWalkTmr)
		

	def Attack(self, npc, target):
		return 10
		
	def Attacked(self, npc, attacker, damage_delt):
		print("Not implemented")
		
# Create an object of our AggressiveNPCBehaviorDefinition
# and assign it to BehaviorDefinition. This is used by the
# server to hook in our behavior.
BehaviorDefinition = AggressiveNPCBehaviorDefinition()
